### DISCUSSION TITLE
Authentication not working for Admin Uid

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#260] [Help Me!] [answered]
Ben Jarrett

I read through #256. Doesn't seem to be the same issue, though. 

Admin is passing as true, but staying as null.
I am getting login and log out buttons, but our admin is staying as null. 
I am expecting my add-projects route to be added to the nav bar when I sign in, but it's bc the admin uid is staying as null. 
I have confirmed that my Admin UID and .ENV files are correct, so I'm at a loss as to why this is happening.

No errors, just logs.

https://github.com/auth0/auth0-react/issues/93
https://stackoverflow.com/questions/60991634/why-when-using-setstate-is-the-console-log-null
https://medium.com/@rivoltafilippo/typing-react-context-to-avoid-an-undefined-default-value-2c7c5a7d5947

I have deleted my admin from firebase and created a new one and added that new uid to my .env file.
I have tried logging the value of the uid.
Double, triple, quadruple checked that I am passing admin everywhere that it needs. 
I checked that all of my proptypes are in ({})
Check that the rest of my crud functionality is working without auth as well. 

https://github.com/BenJarrett/personal-bio-site-react/tree/authentication

**Comment by drteresavasquez** [comment]
There are other tickets about this. Please reference those. And add to those conversations.

Also, did you see the answer to the stack overflow link? That's your answer I believe.

**Reply by drteresavasquez** [reply]
Another thing. Your console warn will show null because updating the state is like a promise. The log will run before the state is set. If that's the only place you see null check your react developer tools, which is what you should be using for state instead of a console warn. If it's not updating in the react dev tools then it's a different issue. 

If your buttons are changing then admin is working properly.

**Reply by BenJarrett** [reply]
Got it. I thought I searched All the tickets and didn’t find one that helped me with this, but I must have missed it . 

I’ll look for the issue ticket and try to understand that stack overflow answer as well.
 
 
Thanks!

**Reply by BenJarrett** [reply]
Just now able to start working on this. 

So my login and log out button are working properly, but my edit and delete buttons are showing whether I am signed and signed out atm

**Reply by drteresavasquez** [reply]
Make sure you're rendering them based on if the user is admin or not. Do you have that logic in the component?

**Reply by drteresavasquez** [reply]
So something like 

```
{admin && <Button />}
```

**Reply by BenJarrett** [reply]
I do. It could be wrong, but I have a function that is set to show the edit and delete button. 
In my map, I am mapping through the card and telling it to show those buttons if it is the admin. 

My edit form and delete functionality were working before I tried to add the authed view for CRUD.

```
const editView = (fbKey) => (
    <div>
      <Button style={{ backgroundColor: '#ffa64d' }} onClick={() => handleClick(fbKey, 'delete')}>Delete Project</Button>
      <Button style={{ backgroundColor: '#004d1a' }} onClick={() => handleClick(fbKey, 'edit')}>
        {editing ? 'Close Form' : 'Edit Project'}
      </Button>
    </div>
  );
  return (
    projects.map((projectInfo) => (
        <Card
        key={projectInfo.firebaseKey}
        style={{
          width: '20rem',
          flex: 'initial',
          flexDirection: 'row',
          flexWrap: 'wrap',
        }} body>
         <CardTitle tag="h5">Project Name: {projectInfo.title}</CardTitle>
         <CardText>url: {projectInfo.url}</CardText>
         <CardText>Github Url: {projectInfo.githubUrl}</CardText>
         <CardText>Technologies Used: {projectInfo.technologiesUsed}</CardText>
         <img style={{ width: '16rem' }} src={projectInfo.screenshot} className="photo" alt="Card image cap" />
         <Button style={{ backgroundColor: '#ffa64d' }} onClick={() => handleClick(projectInfo.firebaseKey, 'delete')}>Delete Project</Button>
      <Button style={{ backgroundColor: '#004d1a' }} onClick={() => handleClick('edit')}>
        {editing ? 'Close Form' : 'Edit Project'}
      </Button>
         { admin && editView(projectInfo.firebaseKey) }
         {
         editing && <ProjectsForm
         formTitle='Edit Project'
         setProjects={setProjects}
         firebaseKey={projectInfo.firebaseKey}
         githubUrl={projectInfo.githubUrl}
         screenshot={projectInfo.screenshot}
         technologiesUsed={projectInfo.technologiesUsed}
         title={projectInfo.title}
         url={projectInfo.url}
         />
         }
         </Card>
    ))
  );
};

**Reply by drteresavasquez** [reply]
Your code is not hiding the buttons. I won't be on again until class but this is something you can research and figure out. Look at the example I gave you above and do more research on it. Look at how many buttons are in this view.

**Comment by BenJarrett** [comment, marked as answer]
I figured this out last night. 
I think my biggest issue is trying to code too fast. 
Most of my errors are due to syntax errors. 

I was passing admin to the add-project route instead of the component being called!
